unity-renderer/Assets/Desktop/Scripts/MainScripts/DCL/Controllers/HUD/Preloading/PreloadingController.cs

```diff
@@ -13,34 +15,40 @@ public class PreloadingController : IDisposable
         private readonly DataStoreRef<DataStore_LoadingScreen> loadingScreenRef;
 
         private BaseVariable<bool> isSignUpFlow => DataStore.i.common.isSignUpFlow;
-        private bool isDisposed = false;
+        private bool isDisposed;
 
-        public PreloadingController()
+        private CancellationTokenSource cancellationTokenSource;
+        public PreloadingController(IAddressableResourceProvider addressableResourceProvider)
         {
-            view = Object.Instantiate(GetView());
+            cancellationTokenSource = new CancellationTokenSource();
+            GetView(cancellationTokenSource.Token).Forget();
+
             loadingScreenRef.Ref.loadingHUD.message.OnChange += OnMessageChange;
             loadingScreenRef.Ref.decoupledLoadingHUD.visible.OnChange += OnDecoupledLoadingScreenVisibilityChange;
             isSignUpFlow.OnChange += SignUpFlowChanged;
-        }
 
-        private void OnDecoupledLoadingScreenVisibilityChange(bool current, bool _)
-        {
-            if (current)
-                Dispose();
-        }
-
-        private GameObject GetView()
-        {
-            return Resources.Load<GameObject>("Preloading");
+            async UniTask GetView(CancellationToken cancellationToken) =>
+                view = (await addressableResourceProvider.Instantiate<Transform>("Preloading", cancellationToken: cancellationToken))
+                   .gameObject;
         }
 
         public void Dispose()
         {
             if (isDisposed) return;
             isDisposed = true;
+
+            cancellationTokenSource.Cancel();
+            cancellationTokenSource.Dispose();
+
             WaitForViewsToFadeOut();
         }
 
+        private void OnDecoupledLoadingScreenVisibilityChange(bool current, bool _)
+        {
+            if (current)
+                Dispose();
+        }
+
         async UniTask WaitForViewsToFadeOut()
```

Comment:
You use `view` there, though it could be `null` if `Addressables.InstantiateAsync(address)` takes too long or fails

Reply:
added null check before the destroy of the view